gatsby-node.js

```diff
@@ -42,6 +42,107 @@ const getMessages = (path, language) => {
   }
 }
 
+/**
+ * Markdown isOutdated check
+ * Parse headers in markdown file (both translated and english) and compare their info structure.
+ * If this structure is not the same, then the file isOutdated.
+ * If there is not english file, return true
+ * @param {string} path filepath for translated mdx file
+ * @returns boolean for if file is outdated or not
+ */
+const checkIsMdxOutdated = (path) => {
+  try {
+    const splitPath = path.split("/")
+    splitPath.splice(7, 2)
+    const englishPath = splitPath.join("/")
+
+    const re = /([#]+) [^\{]+\{#([^\}]+)\}/gim
+
+    const translatedData = fs.readFileSync(path, "utf-8")
+    const englishData = fs.readFileSync(englishPath, "utf-8")
+
+    let englishMatch = ""
+    let intlMatch = ""
+    englishData.match(re).forEach((match) => {
+      englishMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+    translatedData.match(re).forEach((match) => {
+      intlMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+
+    return englishMatch !== intlMatch
+  } catch (err) {
+    return true
```

Comment:
I would make this catch more granular. As you describe in your doc, "If there is not english file, return true", I would lean toward this:
```ts
const splitPath = path.split("/")
splitPath.splice(7, 2)
const englishPath = splitPath.join("/")

try {
  const englishData = fs.readFileSync(englishPath, "utf-8")
} catch (err) {
  return true
}

// rest of the logic
...
```
Because if any other error happens (like not having a match), the consumer is going to be able to handle that, in the way they want.

In this case, we could `console.error` and error if no header id was found.
```ts
try {
  isOutdated = await checkIsMdxOutdated(node.fileAbsolutePath)
} catch (err) {
  // log
  console.error(`No header id found in file ${node.fileAbsolutePath}`, err)
  // OR throw, in case we want to stop the build
  throw new Error(`No header id found in file ${node.fileAbsolutePath}`)
}
```

Reply:
I like the idea of a console error - no need to disrupt the build IMO.